x-pack/plugins/security/public/management/users/user_api_client.ts

```diff
@@ -4,40 +4,39 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { kfetch } from 'ui/kfetch';
-import { Role, User, EditUser } from '../../common/model';
+import { HttpStart } from 'src/core/public';
+import { Role, User, EditUser } from '../../../common/model';
 
 const usersUrl = '/internal/security/users';
 const rolesUrl = '/api/security/role';
 
 export class UserAPIClient {
-  public async getUsers(): Promise<User[]> {
-    return await kfetch({ pathname: usersUrl });
+  constructor(private readonly http: HttpStart) {}
+
+  public async getUsers() {
+    return await this.http.get<User[]>(usersUrl);
   }
 
-  public async getUser(username: string): Promise<User> {
-    const url = `${usersUrl}/${encodeURIComponent(username)}`;
-    return await kfetch({ pathname: url });
+  public async getUser(username: string) {
+    return await this.http.get<User>(`${usersUrl}/${encodeURIComponent(username)}`);
   }
 
   public async deleteUser(username: string) {
-    const url = `${usersUrl}/${encodeURIComponent(username)}`;
-    await kfetch({ pathname: url, method: 'DELETE' }, {});
+    await this.http.delete(`${usersUrl}/${encodeURIComponent(username)}`);
   }
 
   public async saveUser(user: EditUser) {
-    const url = `${usersUrl}/${encodeURIComponent(user.username)}`;
-
-    await kfetch({ pathname: url, body: JSON.stringify(user), method: 'POST' });
+    await this.http.post(`${usersUrl}/${encodeURIComponent(user.username)}`, {
+      body: JSON.stringify(user),
+    });
   }
 
-  public async getRoles(): Promise<Role[]> {
-    return await kfetch({ pathname: rolesUrl });
+  public async getRoles() {
```

Comment:
__note:__ didn't notice that previously, I need to remove these two roles related methods and use `RolesAPIClient` instead I guess.

Reply:
We can tackle this in a followup if you'd rather. I'm flexible

Reply:
Can't resist, will fix it here :)